app/src/main/java/com/example/intandem/fragments/PostsFragment.java

```diff
@@ -164,14 +244,81 @@ public boolean onOptionsItemSelected(@NonNull MenuItem item) {
         return super.onOptionsItemSelected(item);
     }
 
+    private void findCurrentLocation() {
+//        queryPosts();
+        fusedLocationProviderClient = LocationServices.getFusedLocationProviderClient(getActivity());
+
+        // check if gps is enabled or not and then request user to enable it
+        LocationRequest locationRequest = LocationRequest.create();
+        locationRequest.setInterval(10000);
+        locationRequest.setFastestInterval(5000);
+        locationRequest.setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY);
```

Comment:
All of this setup of `localRequest` is basically[ boilerplate code](https://www.freecodecamp.org/news/whats-boilerplate-and-why-do-we-use-it-let-s-check-out-the-coding-style-guide-ac2b6c814ee7/#:~:text=In%20computer%20programming%2C%20boilerplate%20code,code%20to%20do%20minimal%20jobs.)

We can put it all in a helper function like `getLocationRequest()` that does all that somewhere else and keep it out of this method.